### ISSUE TITLE
`PortForwarding(host_port=...)` has no effect

### ISSUE THREAD

**Issue by martinpitt** [dcermak/pytest_container#190]
Selecting a random free host port with `PortForwarding(container_port=N)` in a `Pod` by default makes sense. However, that makes it much more difficult to work with the pod from the host, e.g. our tests produce minio (local S3 container) log URLs like "https://localhost:9000/logs/..." which then don't work (can't be clicked). The user always has to look up the host port in e.g. `podman ps`, and modify the URLs accordingly.

Thus the `PortForwarding` constructor also offers a `host_port` argument. Its [documentation](https://dcermak.github.io/pytest_container/api.html#pytest_container.inspect.PortForwarding.host_port):

> This value is automatically set by the *container_* fixtures, so there’s no need for the user to modify it

That sounds like I *could* set it. But it doesn't have any actual effect.

To demonstrate, create an empty directory, a `test/` subdir in it, and create a `test/test_pod.py` with:
```py
import pytest

from pytest_container import Container, PortForwarding
from pytest_container.pod import Pod, PodData

C_HELLO = Container(url='docker.io/busybox')

MYPOD = Pod(
    containers=[C_HELLO],
    forwarded_ports=[PortForwarding(container_port=9000, host_port=9000)]
)


@pytest.mark.parametrize('pod', [MYPOD], indirect=True)
def test_hello(pod: PodData):
    assert pod.forwarded_ports[0].host_port == 9000
```

This fails with

> FAILED test/test_pod.py::test_hello[pod0] - AssertionError: assert 37369 == 9000

(of course the actual number differs between each run)

**Comment by dcermak** [comment]
> > This value is automatically set by the _container__ fixtures, so there’s no need for the user to modify it
> 
> That sounds like I _could_ set it. But it doesn't have any actual effect.

Technically, the documentation doesn't claim that the value of `host_port` made by the user will be honored, but  I do agree that it could be less ambiguous.

The main issue with setting host ports, is that you absolutely must be certain, that the a) this container/pod will not be launched a second time and b) no other container will try to open this port.

Let me see if I can implement it in a satisfactory fashion.

**Comment by dcermak** [comment]
I cannot reproduce the behavior you're facing on the `main` branch anymore. Judging from the git blame, it has been fixed in 505bc380348d8c66258486c235a1adfc8438d403 but that has not been part of the 0.3.0 released yet.

Could you retry from the main branch please, just to be sure?

**Comment by martinpitt** [comment]
@dcermak Awesome, :heart: retroactive bug fixing :grin: 
With `sudo pip install git+https://github.com/dcermak/pytest_container` this indeed works. I'll also use this in my test to get rid of the nasty `extra_entrypoint_args` workaround.

Thanks!
